8.15.4 Release notes (#6062)
elastic/security-docs: docs/release-notes/8.15.asciidoc

[[release-notes-header-8.15.0]]
== 8.15

[discrete]
[[release-notes-8.15.4]]
=== 8.15.4

[discrete]
[[known-issue-8.15.4]]
==== Known issues

// tag::known-issue-189676[]
[discrete]
.Tags appear in Elastic AI Assistant's responses
[%collapsible]
====
*Details* +
On August 1, 2024, it was discovered that Elastic AI Assistant's responses when using Bedrock Sonnet 3.5 may include `<antThinking>` tags, for example `<search_quality_reflection>` ({kibana-issue}189676[#189676]).

====
// end::known-issue-189676[]

[discrete]
[[enhancements-8.15.4]]
==== Enhancements
* Enhances {elastic-defend} by improving the `call_stack_final_user_module` attribution where potential `proxy_call` modules are encountered during Windows call stack analysis.

[discrete]
[[bug-fixes-8.15.4]]
==== Bug fixes

> * Fixes a bug that caused an Elastic AI Assistant error if you had over 20 conversations and tried to access or update any of them ({kibana-pull}197305[#197305]).

Reviewer: ```suggestion
* Fixes a conflict that can result in a Windows boot failure `0xC000007B` for `ElasticElam.sys` when Elastic Defend 8.15.2 or 8.15.3 is installed alongside CrowdStrike.
* Fixes a bug that caused an Elastic AI Assistant error if you had over 20 conversations and tried to access or update any of them ({kibana-pull}197305[#197305]).
```

[[release-notes-header-8.15.0]]
== 8.15

[discrete]
[[release-notes-8.15.4]]
=== 8.15.4

[discrete]
[[known-issue-8.15.4]]
==== Known issues

// tag::known-issue-189676[]
[discrete]
.Tags appear in Elastic AI Assistant's responses
[%collapsible]
====
*Details* +
On August 1, 2024, it was discovered that Elastic AI Assistant's responses when using Bedrock Sonnet 3.5 may include `<antThinking>` tags, for example `<search_quality_reflection>` ({kibana-issue}189676[#189676]).

====
// end::known-issue-189676[]

[discrete]
[[enhancements-8.15.4]]
==== Enhancements
* Enhances {elastic-defend} by improving the `call_stack_final_user_module` attribution where potential `proxy_call` modules are encountered during Windows call stack analysis.

[discrete]
[[bug-fixes-8.15.4]]
==== Bug fixes

> * Fixes a conflict that can result in a Windows boot failure `0xC000007B` for `ElasticElam.sys` when {elastic-defend} 8.15.2 or 8.15.3 is installed alongside CrowdStrike.

Reviewer 2: ```suggestion
* Fixes a conflict that could result in a Windows boot failure `0xC000007B` for `ElasticElam.sys` when {elastic-defend} 8.15.2 or 8.15.3 was installed alongside CrowdStrike.
```

* Fixes a bug that caused an Elastic AI Assistant error if you had over 20 conversations and tried to access or update any of them ({kibana-pull}197305[#197305]).
* Makes Automatic Import more forgiving if LLMs return ECS mappings in unexpected formats ({kibana-pull}195167[#195167]).

> * Fixes a bug that caused fields from all indices to display when adding a filter to a rule that you were editing. Now, only fields from the rule's specified indices appear ({kibana-pull}194678[#194678]).

Reviewer 3: same PR fixed also this issue: https://github.com/elastic/kibana/issues/181643
Would it be worth to mention it?

Author: Sure! I can reference 181643 in this summary as well.

```suggestion
* Fixes a bug that caused fields from all indices to display when adding a filter to a rule that you were editing. Now, only fields from the rule's specified indices appear ({kibana-pull}194678[#194678], {kibana-pull}181643[#181643]).
```

Reviewer 3: @nastasha-solomon , there is no doc instance to check change in this PR, but https://github.com/elastic/kibana/issues/181643 is issue. Does it work well `{kibana-pull}181643[#181643]` as pull request?

Author: Hey @vitaliidm, just saw this now. We don't typically link to issues in the fixed bug section of the release notes, but it's possible to do so if needed. I'll open a new PR to fix this reference.

Author: This should be fixed in https://github.com/elastic/security-docs/pull/6126. Thanks for reporting this @vitaliidm!

* Improves {elastic-defend} by making the `elastic-endpoint status` command more reliable. Before this fix, the command occasionally failed with an I/O error.
* Fixes an {elastic-defend} process crash that could occur if it was configured to use the Kafka output.

> * Fixes a bug where {elastic-defend} could fail to properly enrich Windows API events for short-lived processes on older operating systems that don't natively include this telemetry, such as Windows Server 2019. This might result in dropped or unattributed API events.  

Reviewer 2: ```suggestion
* Fixes a bug where {elastic-defend} could fail to properly enrich Windows API events for short-lived processes on older operating systems that didn't natively include this telemetry, such as Windows Server 2019. This could result in dropped or unattributed API events.  
```

> * Ensures that {elastic-defend} does not emit an empty `memory_region` if it can't enrich a memory region in an API event. With this fix, {elastic-defend} removes these fields.

Reviewer 2: ```suggestion
* Ensures that {elastic-defend} does not emit an empty `memory_region` if it can't enrich a memory region in an API event. After this fix, {elastic-defend} removes these fields.
```

> * Fixes an {elastic-defend} bug where Windows API events might be dropped if they contain Unicode characters that can't be converted to ANSI.

Reviewer 2: ```suggestion
* Fixes an {elastic-defend} bug where Windows API events could be dropped if they contained Unicode characters that couldn't be converted to ANSI.
```
